packages/framework/aqueduct/src/data-object-factories/dataObjectFactory.ts

```diff
@@ -55,6 +70,85 @@ export class DataObjectFactory<
 			mergedObjects.push(SharedMap.getFactory());
 		}
 
-		super(type, ctor, mergedObjects, optionalProviders, registryEntries, runtimeFactory);
+		let converted = false;
+		const convertRoundTripP = new Deferred<void>();
+
+		const fullConvertDataFn =
```

Comment:
Is there a way to move most of this logic out of DataObjectFactory into derived class?
Mostly for readability.